spinalcordtoolbox/reports/slice.py

```diff
@@ -183,21 +183,19 @@ def add_slice(matrix, i, column, patch):
         return matrix
 
     @staticmethod
-    def nan_fill(A):
-        """Interpolate NaN values with neighboring values in array (in-place)
-        If only NaNs, return an array of zeros.
+    def inf_nan_fill(A):
+        """Interpolate inf and NaN values with neighboring values in a 1D array, in-place.
+        If only inf and NaNs, fills the array with zeros.
         """
-        nans = np.isnan(A)
-        if ~np.any(nans):
-            return A
-        elif np.all(nans):
-            A[:] = np.zeros_like(A)
-            return A
-        xp = (~nans).ravel().nonzero()[0]
-        fp = A[~nans]
-        x = nans.ravel().nonzero()[0]
```

Comment:
I noticed that the `.ravel()` calls were removed. Is this because `interp` only works on 1D arrays anyway, so there's already an implicit expectation that `A` will be 1D, making the `ravel` calls redundant? (Since, I presume the old versions would have failed anyway had `A` been 2+D, with or without the `ravel` calls.)

Reply:
That's right, `np.interp` only works with 1D arrays, and `inf_nan_fill` also only makes sense for 1D arrays (now mentioned in the docstring!), so there's no need to convert between 1D and 1D arrays.

(As an aside, `np.ravel` and `np.ndarray.flat` can be useful but they are often a sign of something unsafe happening, so I try to avoid them. Or at least to make sure that I don't mix flattened indices with 1D indices.)